core/commands/partials/vagrant_configuration_generator.rb

```diff
@@ -329,12 +209,12 @@ def parse_products_info(node)
   def node_definition(node, path, cookbook_path)
```

Comment:
This method calls AttributeGenerator too many times. Can you provide a clearer alternative?

The same goes for the TerraformConfigurationGenerawor#generate_node_info method.